pkg/project/registry/projectrequest/delegated/delegated_test.go

```diff
@@ -0,0 +1,88 @@
+package delegated
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	rbaclisters "k8s.io/kubernetes/pkg/client/listers/rbac/internalversion"
+
+	"github.com/go-openapi/errors"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+)
+
+func TestDelegatedWait(t *testing.T) {
+	cache := &testRoleBindingLister{}
+	storage := &REST{roleBindings: cache}
+
+	cache.namespacelister = &testRoleBindingNamespaceLister{}
+	cache.namespacelister.bindings = map[string]*rbac.RoleBinding{}
+	cache.namespacelister.bindings["anything"] = nil
+
+	callReturnedCh := testWait(storage)
+
+	select {
+	case <-callReturnedCh:
+		t.Errorf("too fast, should have blocked")
```

Comment:
I'd use `t.Error()` here and rephrase the error message to provide more details (like what happens, what is expected, waiting time).

I just imagined myself, if I would found `too fast, should have blocked` error somewhere in the long test execution log, then I'd like to have more details.